pkgs/misc/scrcpy/default.nix

```diff
@@ -0,0 +1,51 @@
+{ stdenv, fetchurl, fetchFromGitHub, makeWrapper
+, meson
+, ninja
+, pkgconfig
+
+, androidsdk
+, ffmpeg
+, SDL2
+}:
+
+let
+  version = "1.2";
+  prebuilt_server = fetchurl {
+    url = "https://github.com/Genymobile/scrcpy/releases/download/v${version}/scrcpy-server-v${version}.jar";
+    sha256 = "0q0zyqw7y33r9ybjp8ay6yac7ifca1lq14pjvw6x78zxs976affb";
+  };
+in
+stdenv.mkDerivation rec {
+  name = "scrcpy-${version}";
+  inherit version;
+  src = fetchFromGitHub {
+    owner = "Genymobile";
+    repo = "scrcpy";
+    rev = "v${version}";
+    sha256 = "01zw6h6mz2cwqfh9lwypm8pbfx9m9df91l1fq1i0f1d8v49x8wqc";
+  };
+
+  nativeBuildInputs = [ makeWrapper meson ninja pkgconfig];
+
+  buildInputs = [ ffmpeg SDL2 ];
+
+  # Manually install the server jar to prevent Meson from "fixing" it
+  preConfigure = ''
+    echo -n > server/meson.build
```

Comment:
I manually install it to work around the [weird elf error from Meson](https://gist.github.com/lukeadams/ab2877b7137a7ce25b3a035525047b41). I don't really have experience with meson and couldn't find a way to exclude the file from the [depfixer.py](https://github.com/mesonbuild/meson/blob/master/mesonbuild/scripts/depfixer.py) invocation.

depfixer has a `fix_jar` function, but due to a bug (???) it calls fix_rpath instead.

Reply:
I have seen the same issue. The prebuild server should be fine.

Reply:
This is very strange, I didn't have this issue when i was creating the package because I was at an old version of nixpkgs, I think this is the new version of meson https://github.com/mesonbuild/meson/commit/c70a051e9352639440532ad969d9b7c215a3e699 Perhaps this commit

Reply:
Aw yes it's this commit, meson try to rewrite the file to rewrite the Manifest

Reply:
Hmm so the version in nixpkgs (0.46.1) doesn't have [fix_jar](https://github.com/mesonbuild/meson/blob/0.46.1/mesonbuild/scripts/depfixer.py#L288). Updating meson to 0.47.1 may fix it

Reply:
Aw we we are not at 0.47.1 :thinking:, even if we update I think we still will have issues because with 0.47.1 it will try to rewrite the jarfile, and fail like `fix_rpath` because the file is not writtable